### DISCUSSION TITLE
mcp960x Address Range for

### DISCUSSION THREAD

**Discussion by jeffxor** [UncleRus/esp-idf-lib#318] [Q&A] [answered]
I am working with the following [MCP9600 I2C THERMOCOUPLE AMP](https://www.digikey.com/en/products/detail/4101/1528-4101-ND/10245127?itemSeq=324782003) which has a default address of 0x65 and the ability to set to 0x67.

I am wondering why the following was added to restrict the address range for this module? I have commented out the check to use the address I need and it appears everything works fine. 

I am not sure what I should do to move forward. I am willing to open a new PR request, but not sure how to handle my needs. Comment our the code or create a new module explicit for the  MCP9600.

https://github.com/UncleRus/esp-idf-lib/blob/a43fa649684a058d69c7b1a739baf090294fc1ff/components/mcp960x/mcp960x.c#L175

**Comment by UncleRus** [comment]
Hi!
This is common practice for all drivers in this library. This approach allows to identify common errors in programs associated with incorrect device addresses.
If you find that the driver does not support a valid address, please let me know this address and I will add it to the condition.

**Comment by jeffxor** [comment]
Hi!,

Yes. As I mentioned I have the following MCP9600 I2C THERMOCOUPLE AMP
<https://www.digikey.com/en/products/detail/4101/1528-4101-ND/10245127?itemSeq=324782003>
it
allows for addresses (0x65 and 0x67). it would be great to add those

**Comment by UncleRus** [comment, marked as answer]
Fixed in ccdacdd.
Thank you!
